samples/snippets/fsharp/azure/blob-storage.fsx

```diff
@@ -130,40 +111,22 @@ let ListBlobsSegmentedInFlatListing(container:CloudBlobContainer) =
         // When the continuation token is null, the last page has been 
         // returned and execution can exit the loop.
 
-        let rec loop continuationToken (i:int) = 
-            async {
-                let! ct = Async.CancellationToken
-                // This overload allows control of the page size. You can return
-                // all remaining results by passing null for the maxResults 
-                // parameter, or by calling a different overload.
-                let! resultSegment = 
-                    container.ListBlobsSegmentedAsync(
-                        "", true, BlobListingDetails.All, Nullable 10, 
-                        continuationToken, null, null, ct) 
-                    |> Async.AwaitTask
-
-                if (resultSegment.Results |> Seq.length > 0) then
-                    printfn "Page %d:" i
-
-                for blobItem in resultSegment.Results do
-                    printfn "\t%O" blobItem.StorageUri.PrimaryUri
-
-                printfn ""
-
-                // Get the continuation token.
-                let continuationToken = resultSegment.ContinuationToken
-                if (continuationToken <> null) then
-                    do! loop continuationToken (i+1)
-            }
-    
-        do! loop null 1
+        let blobs = new List<BlobItem>()
+
+        for page in container.GetBlobs().AsPages() do
+            blobs.AddRange(page.Values)
+
+        for blobItem in blobs do 
+            printf "The BlobItem is : %s " blobItem.Name
+        printfn ""
+
     }
 
 // Create some dummy data by uploading the same file over and over again
 for i in 1 .. 100 do
-    let blob  = container.GetBlockBlobReference("myblob" + string i + ".txt")
+    let blob  = container.GetBlobClient("myblob" + string i + ".txt")
```

Comment:
We should consider updating this to use string interpolation.  _([ref](https://docs.microsoft.com/dotnet/fsharp/language-reference/interpolated-strings))_